doc/source/configuration.rst

```diff
@@ -0,0 +1,270 @@
+according to your credentials.
+
+.. code-block:: bash
+
+    # .ostackrc file
+    export OS_USERNAME="<USER_NAME>"
+    export OS_USER_DOMAIN_NAME=<OTC00000000001000000XYZ>
+    export OS_PASSWORD=<PASSWORD> # optional
+    export OS_TENANT_NAME=eu-de
+    export OS_PROJECT_NAME=<eu-de_PROJECT_NAME>
+    export OS_AUTH_URL=https://iam.eu-de.otc.t-systems.com:443/v3
+    export NOVA_ENDPOINT_TYPE=publicURL
+    export OS_ENDPOINT_TYPE=publicURL
+    export CINDER_ENDPOINT_TYPE=publicURL
+    export OS_VOLUME_API_VERSION=2
+    export OS_IDENTITY_API_VERSION=3
+    export OS_IMAGE_API_VERSION=2
+
+Run the source command to make the ``environment variables`` available.
+
+.. code-block:: bash
+
+   $ source .ostackrc
+
+The ``environment variables`` are now available for usage. For testing your
+connection run the following command.
+
+Test your connection
+^^^^^^^^^^^^^^^^^^^^
+
+If you followed the `installation advices <index>`_ for your specific
+operating system, you can use the following command to test the basic
+functionality.
+
+.. code-block:: bash
+
+    $ openstack flavor list
+
+.. note::
+   You don't need to specify the `--os-cloud` parameter when environment
+   variables are used.
+
+Cache Settings
+--------------
+
+Accessing a cloud is often expensive, so it's quite common to want to do some
+client-side caching of those operations. To facilitate that, `openstacksdk`
+understands passing through cache settings to dogpile.cache, with the following
+behaviors:
+
+* Listing no config settings means you get a null cache.
+* `cache.expiration_time` and nothing else gets you memory cache.
+* Otherwise, `cache.class` and `cache.arguments` are passed in
+
+Different cloud behaviors are also differently expensive to deal with. If you
+want to get really crazy and tweak stuff, you can specify different expiration
+times on a per-resource basis by passing values, in seconds to an expiration
+mapping keyed on the singular name of the resource. A value of `-1` indicates
+that the resource should never expire.
+
+`openstacksdk` does not actually cache anything itself, but it collects
+and presents the cache information so that your various applications that
+are connecting to OpenStack can share a cache should you desire.
+
+.. code-block:: yaml
+
+  cache:
+    class: dogpile.cache.pylibmc
+    expiration_time: 3600
+    arguments:
+      url:
+        - 127.0.0.1
+    expiration:
+      server: 5
+      flavor: -1
+  clouds:
+    mtvexx:
+      profile: otc
+      auth:
+        username: "<USER_NAME>"
+        password: '<PASSWORD>'
+        project_name: <eu-de_PROJECT_NAME>
+      region_name: eu-de
+      dns_api_version: 1
+
+`openstacksdk` can also cache authorization state (token) in the keyring.
+That allow the consequent connections to the same cloud to skip fetching new
+token. When the token gets expired or gets invalid `openstacksdk` will
+establish new connection.
+
+
```

Comment:
can you please also include here the MFA configuration case (https://docs.openstack.org/openstacksdk/latest/user/config/configuration.html#mfa-support)?

Reply:
added mfa